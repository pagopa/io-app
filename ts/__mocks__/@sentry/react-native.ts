export const captureException = jest.fn();
export const captureMessage = jest.fn();
export const captureEvent = jest.fn();
export const setContext = jest.fn();
export const setExtras = jest.fn();
export const setExtra = jest.fn();
export const setTags = jest.fn();
export const setTag = jest.fn();
export const setUser = jest.fn();
export const lastEventId = jest.fn();
export const captureCheckIn = jest.fn();
export const withMonitor = jest.fn();
export const flush = jest.fn().mockResolvedValue(false);
export const close = jest.fn().mockResolvedValue(false);
export const isInitialized = jest.fn().mockReturnValue(false);
export const isEnabled = jest.fn().mockReturnValue(false);
export const addEventProcessor = jest.fn();
export const startSession = jest.fn();
export const endSession = jest.fn();
export const captureSession = jest.fn();
export const ErrorBoundary = jest.fn(({ children }) => children);
