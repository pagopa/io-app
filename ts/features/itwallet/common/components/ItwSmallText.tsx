import { IOToast, LabelSmall } from "@pagopa/io-app-design-system";
import React, { ReactNode } from "react";
import Markdown, { ASTNode } from "react-native-markdown-display";
import I18n from "../../../../i18n";
import { openWebUrl } from "../../../../utils/url";

type ItwSmallTextProps = {
  children?: string;
};

const rules = {
  paragraph: (node: ASTNode, children: Array<ReactNode>) => (
    <LabelSmall
      key={node.key}
      weight={"Regular"}
      color={"bluegrey"}
      testID="itwSmallTextParagraph"
    >
      {children}
    </LabelSmall>
  ),
  link: (node: ASTNode, children: Array<ReactNode>) => (
    <LabelSmall
      key={node.key}
      color={"blue"}
      onPress={() =>
        openWebUrl(node.attributes.href, () =>
          IOToast.error(I18n.t("global.jserror.title"))
        )
      }
      numberOfLines={1}
      testID="itwSmallTextLink"
    >
      {children}
    </LabelSmall>
  )
};

/**
 * This component renders a small text, with optional link, using react components starting from a markdown text.
 * Main components are mapped into rules object.
 * @param children - contains the text to be converted in react elements.
 */
export const ItwSmallText: React.FC<ItwSmallTextProps> = ({ children }) => (
  <Markdown rules={rules}>{children}</Markdown>
);
