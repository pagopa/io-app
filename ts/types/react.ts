import { Component, ComponentType, StatelessComponent } from "react";
import {
  NavigationInjectedProps,
  NavigationScreenProps
} from "react-navigation";

/**
 * Evaluates to the Props type of a React component
 */
export type ComponentProps<C> = C extends StatelessComponent<infer P1>
  ? P1
  : C extends Component<infer P2>
    ? P2
    : C extends ComponentType<infer P3> ? P3 : never;

/**
 * Infers the type of the navigation params of a component
 */
export type InferNavigationParams<
  C,
  P = ComponentProps<C>
> = P extends NavigationInjectedProps<infer N>
  ? N
  : P extends NavigationScreenProps<infer N1> ? N1 : never;
