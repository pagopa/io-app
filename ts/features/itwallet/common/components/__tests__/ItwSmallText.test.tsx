import * as React from "react";
import { render } from "@testing-library/react-native";
import { ItwSmallText } from "../ItwSmallText";

describe("ItwSmallText", () => {
  it(`should match snapshot`, () => {
    const component = render(
      <ItwSmallText>
        Per maggiori informazioni, leggi l’[informativa Privacy]() e i [Termini
        e Condizioni d’uso]()
      </ItwSmallText>
    );
    expect(component).toMatchSnapshot();
  });
});
