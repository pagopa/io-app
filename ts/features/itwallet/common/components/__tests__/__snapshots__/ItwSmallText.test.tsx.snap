// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItwSmallText should match snapshot 1`] = `
<View
  style={Object {}}
>
  <Text
    allowFontScaling={false}
    color="bluegrey"
    defaultColor="blue"
    defaultWeight="Bold"
    font="TitilliumSansPro"
    fontStyle={
      Object {
        "fontSize": 14,
        "lineHeight": 21,
      }
    }
    style={
      Array [
        Object {
          "fontSize": 14,
          "lineHeight": 21,
        },
        Object {
          "color": "#475A6D",
          "fontFamily": "Titillium Sans Pro",
          "fontStyle": "normal",
          "fontWeight": "400",
        },
      ]
    }
    testID="itwSmallTextParagraph"
    weight="Regular"
  >
    <Text
      style={Object {}}
    >
      <Text
        style={
          Array [
            Object {},
            Object {},
          ]
        }
      >
        Per maggiori informazioni, leggi l’
      </Text>
      <Text
        allowFontScaling={false}
        color="blue"
        defaultColor="blue"
        defaultWeight="Bold"
        font="TitilliumSansPro"
        fontStyle={
          Object {
            "fontSize": 14,
            "lineHeight": 21,
          }
        }
        numberOfLines={1}
        onPress={[Function]}
        style={
          Array [
            Object {
              "fontSize": 14,
              "lineHeight": 21,
            },
            Object {
              "color": "#0073E6",
              "fontFamily": "Titillium Sans Pro",
              "fontStyle": "normal",
              "fontWeight": "700",
            },
          ]
        }
        testID="itwSmallTextLink"
        weight="Bold"
      >
        <Text
          style={
            Array [
              Object {
                "textDecorationLine": "underline",
              },
              Object {},
            ]
          }
        >
          informativa Privacy
        </Text>
      </Text>
      <Text
        style={
          Array [
            Object {},
            Object {},
          ]
        }
      >
         e i 
      </Text>
      <Text
        allowFontScaling={false}
        color="blue"
        defaultColor="blue"
        defaultWeight="Bold"
        font="TitilliumSansPro"
        fontStyle={
          Object {
            "fontSize": 14,
            "lineHeight": 21,
          }
        }
        numberOfLines={1}
        onPress={[Function]}
        style={
          Array [
            Object {
              "fontSize": 14,
              "lineHeight": 21,
            },
            Object {
              "color": "#0073E6",
              "fontFamily": "Titillium Sans Pro",
              "fontStyle": "normal",
              "fontWeight": "700",
            },
          ]
        }
        testID="itwSmallTextLink"
        weight="Bold"
      >
        <Text
          style={
            Array [
              Object {
                "textDecorationLine": "underline",
              },
              Object {},
            ]
          }
        >
          Termini e Condizioni d’uso
        </Text>
      </Text>
    </Text>
  </Text>
</View>
`;
